addons/sale/static/src/js/product_discount_field.js

```diff
@@ -0,0 +1,32 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+import { FloatField } from "@web/views/fields/float/float_field";
+import { _lt } from "@web/core/l10n/translation";
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+
+import { updateAll } from "./update_all_widget";
+
+
+export class ProductDiscountField extends FloatField {
+    setup() {
+        super.setup();
+        this.dialogService = useService("dialog");
+    }
+
+    comparisonFunction(arg1, arg2) {
+        return arg1 === arg2;
+    }
+
+    onChange(ev) {
+        // Call to update all lines.
+        updateAll(this, ev.target.value, "discount", this.comparisonFunction);
```

Comment:
you will be able to access this list data with `this.env.model.root.data.{field_name_of_list}`
and pass it to the updateAll function